websocket/websocket.go

```diff
@@ -0,0 +1,116 @@
+package websocket
+
+import (
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+
+	log "github.com/cihub/seelog"
+	"github.com/gobwas/ws"
+	"github.com/gobwas/ws/wsutil"
+	socketio "github.com/googollee/go-socket.io"
+	"github.com/julienschmidt/httprouter"
+)
+
+const serviceUpdateStatus = "server/SERVICE_UPDATE_STATUS"
+
+// Instance of websocket server
+var Instance WebSocket
+
+// WebSocket struct
+type WebSocket struct {
+	Server      *socketio.Server
+	connections map[*net.Conn]bool
+	actions     chan action
+}
+
+// ServiceUpdateStatusAction - send new service instance status to websocket
+func (webSocket WebSocket) ServiceUpdateStatusAction(payload interface{}) {
```

Comment:
You'll want to use a pointer receiver here. Otherwise you're pushing to a copy of the channel, not the channel you want to push to, see https://play.golang.org/p/KrNAwzLLWt6